docs: enforce jsdoc in headless + stricter jsdoc rules in core

quentinderoubaix opened the pull request.

fbasso commented on core/src/services/transitions/cssTransitions.ts:
```diff
@@ -26,6 +26,12 @@ export function getTransitionDurationMs(element: HTMLElement): number {
+/**
+ * Create a simple css transition.
+ *
+ * @param start - a function that creates the css animation and returns a clean-up function
```
`@param start - {@link CSSTransitionFn} function that creates the css animation and returns a clean-up function`

fbasso commented on core/src/services/transitions/simpleClassTransition.ts:
```diff
@@ -31,14 +31,20 @@ export interface SimpleClassTransitionContext {
+ * `animationPendingShowClasses` and `animationPendingHideClasses` are attached when transitionning towards one direction
+ * `showClasses` and `hideClasses` are attached when the transition has reached the show or hide state respectively
+ *
+ * @param config - the transition config
```
`@param config - {@link SimpleClassTransitionConfig} config`

fbasso commented on core/src/utils/internal/dom.ts:
```diff
@@ -36,11 +36,23 @@ export function reflow(element: HTMLElement = document.body) {
+ * Attach the given css classes to the element
+ *
+ * @param element - the HTML element
+ * @param classes - the css lcasses
```
lcasses => classes

fbasso commented on core/src/utils/internal/promise.ts:
```diff
@@ -120,6 +171,12 @@ export const promiseFromEvent = (element: EventTarget, event: string) => {
+ *
+ * @param delay - the delay in milli seconds
+ * @returns a promise and an unsubscribe function
+ */
```
Probably examples would be useful for these promise utilities, but maybe will see it in another PR (when the doc will be generated)

fbasso commented on core/src/utils/internal/scrollbars.ts:
```diff
@@ -20,11 +20,17 @@ const internalRemoveScrollbars = () => {
 let internalRevert = noop;
 
+/**
+ * A function to remove the scrollbars on the body element. It can be reverted using the `revertScrollbars` function.
```
` A function to remove the scrollbars on the body element. It can be reverted using the {@link revertScrollbars} function.`

fbasso commented on core/src/utils/internal/scrollbars.ts:
```diff
@@ -20,11 +20,17 @@ const internalRemoveScrollbars = () => {
 };
 
+/**
+ * A function to revert the removal of scrollbars performed by the `removeScrollbars` function.
```
`A function to revert the removal of scrollbars performed by the {@link removeScrollbars} function.`

fbasso commented on core/src/utils/internal/sort.ts:
```diff
@@ -1,5 +1,19 @@
+/**
+ * The default comparision between two inputs, using the javascript < and > signs.
```
inputs => values (?)

fbasso commented on core/src/utils/internal/sort.ts:
```diff
@@ -1,5 +1,19 @@
 export const compareDefault = (a: any, b: any) => (a < b ? -1 : a > b ? 1 : 0);
 
+/**
+ * A comparision function between DOM elements, based on `Node.compareDocumentPosition`.
```
It can be useful to have the link : `A comparision function between DOM elements, based on [Node.compareDocumentPosition](https://developer.mozilla.org/fr/docs/Web/API/Node/compareDocumentPosition).`

fbasso commented on core/src/utils/stores.ts:
```diff
@@ -248,6 +264,12 @@ export const writablesForProps = <T extends object>(
 };
 
+/**
+ * Using input stores, builds an object containing the stores as readable and a global state.
```
Maybe `Using input stores, this function builds an object containing the stores as readable and a global state.`

fbasso reviewed (commented):
Thanks for your PR @quentinderoubaix !

I've added some comments. I think what it misses is examples in the different utilities (for a better understanding), but it can be done once the doc will be generated, it will be easier to see what is missing.

quentinderoubaix commented on angular/headless/src/utils/stores.ts:
```diff
@@ -5,6 +5,14 @@ import {ZoneWrapper} from './zone';
+/**
+ * Convert a tansu readable signal into an Angular signal.
+ *
+ * Note that as it uses Angular's `inject`, this can only be called at component construction time.
```
typedoc indeed displays differently with @remarks, great idea !

quentinderoubaix commented on angular/headless/src/utils/widget.ts:
```diff
@@ -35,6 +35,17 @@ const createPatchSlots = <T extends object>(set: (object: Partial<T>) => void) =
+/**
+ * Call a widget factory using provided configs.
+ *
+ * @param inputs - the inputs
```
Indeed, thanks for noticing !

quentinderoubaix commented:
> Thanks for your PR @quentinderoubaix !
> 
> I've added some comments. I think what it misses is examples in the different utilities (for a better understanding), but it can be done once the doc will be generated, it will be easier to see what is missing.

Thx for the review @fbasso !
I have updated the PR following your different propositions.
I agree that with typedoc online, it would be easier to track what we are missing.

fbasso reviewed (approved):
LGTM on my side for this version ! 
We'll improve things when visual docs for these parts is available.
